handlers/routeservice_test.go

```diff
@@ -306,6 +306,32 @@ var _ = Describe("Route Service Handler", func() {
 				})
 			})
 
+			Context("when a request has a valid route service signature and metadata header and URL contains special chars", func() {
+				BeforeEach(func() {
+					reqArgs, err := config.Request("", "https://my_host.com/resource+9-9_9?query=%23%25")
+					Expect(err).ToNot(HaveOccurred())
+					req.Header.Set(routeservice.HeaderKeySignature, reqArgs.Signature)
+					req.Header.Set(routeservice.HeaderKeyMetadata, reqArgs.Metadata)
+				})
+
+				It("strips headers and sends the request to the backend instance", func() {
+					handler.ServeHTTP(resp, req)
+
+					Expect(resp.Code).To(Equal(http.StatusTeapot))
```

Comment:
It seems like everything passed this code test could be removed. This is really what we are testing. All the rest if just the same as the test just above. We don't need to test again that it "strips headers and sends the request to the backend instance", but just that there are no errors.

Reply:
My idea for the other checks was that we assure that we were not breaking anything from the test above when using the special sequence of characters. But I can see why it is test duplication, therefore I'll remove the other checks.

Reply:
Hi, I have updated the PR so now the test doesn't test again the same as the above, but only the necessary.